go/ct/st/transient.go

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the GNU Lesser General Public Licence v3
+//
+
+package st
+
+import (
+	. "github.com/Fantom-foundation/Tosca/go/ct/common"
+	"golang.org/x/exp/maps"
+)
+
+type Transient struct {
+	storage map[U256]U256
+}
+
+func (t *Transient) SetStorage(key U256, value U256) {
+	if t.storage == nil {
+		t.storage = make(map[U256]U256)
+		t.storage[key] = value
+	} else {
+		t.storage[key] = value
+	}
+}
+
+func (t *Transient) IsSet(key U256) bool {
+	_, ok := t.storage[key]
+	return ok
+}
+
+func (t *Transient) IsNotSet(key U256) bool {
+	return !t.IsSet(key)
+}
+
+func (t *Transient) GetStorage(key U256) U256 {
+	return t.storage[key]
+}
+
+func (t *Transient) DeleteStorage(key U256) {
```

Comment:
This should have the same effect as `SetStorage(key, Value{})`.

In the semantic of the transient storage (or the regular Storage), all keys actually always have a value. It is just that most of them are zero and not explicitly stored. However, logically they are all always present. They can not be created nor deleted.

It is more accurate to think about those as a huge array with 2^256 entries where most of them are zero. You would also not define an operation on the array deleting an index position. This is not really possible.